compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -1223,9 +1229,14 @@ public void visit(BLangQueryExpr queryExpr) {
 
     @Override
     public void visit(BLangFromClause fromClause) {
-        populateUnusedVariableMapForMembers(this.unusedLocalVariables,
-                                            (BLangVariable) fromClause.variableDefinitionNode.getVariable());
-        analyzeNode(fromClause.collection, env);
+        BLangExpression collection = fromClause.collection;
+
+        if (isNotRangeExpr(collection)) {
```

Comment:
Not sure, we need to allow this here. In the issue, we discussed only to allow for `foreach` stmt, because it is simple use case.

Reply:
Raised this in the spec issue - https://github.com/ballerina-platform/ballerina-spec/issues/356#issuecomment-964983420.